src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -114,13 +115,13 @@ object ChannelOwner {
       }
       case request@CreateConsumer(listener) => {
         log.debug(s"creating new consumer for listener $listener")
-        sender ! withChannel(channel, request)(c => new DefaultConsumer(channel) {
-          override def handleDelivery(consumerTag: String, envelope: Envelope, properties: BasicProperties, body: Array[Byte]) {
+        sender ! withChannel(channel, request)(_ => new DefaultConsumer(channel) {
+          override def handleDelivery(consumerTag: String, envelope: Envelope, properties: BasicProperties, body: Array[Byte]): Unit = {
             listener ! com.github.sstone.amqp.Amqp.Delivery(consumerTag, envelope, properties, body)
           }
         })
       }
-      case request@ConfirmSelect => {
+      case request: ConfirmSelect.type => {
```

Comment:
Huh? Do we need it because of some warning?

Reply:
Yes,

```
[warn] amqp-client/src/main/scala/com/github.sstone/amqp/ChannelOwner.scala:124:20: The value matched by com.github.sstone.amqp.Amqp.ConfirmSelect is bound to request, which may be used under the
[warn] unsound assumption that it has type com.github.sstone.amqp.Amqp.ConfirmSelect.type, whereas we can only safely
[warn] count on it having type Any, as the pattern is matched using `==` (see scala/bug#1503).
[warn]       case request@ConfirmSelect => {
[warn]                    ^
```

In 2.13 it casts `request` to `Any` and call of `withChannel()` doesn't compile because it requires a subtype of `Request`.